### ISSUE TITLE
Elasticsearch Language [Greek]

### ISSUE THREAD

**Issue by OlgaLom** [10up/ElasticPress#3762]
### Describe your question

Hello, 

I have elasticpress on my wp site. In the setting, I selected Greek as the language. 
But what does that setting actually do?

When I write on the search something without a tone e.g. " Darphin κρεμες"  it doesn't return anything but if I search "Darphin κρέμες" it returns some products.

I know that the letters with tones are translated with different ASCI characters but I thought that the language setting in the plugin was gonna fix these "issues". 

Should I add some custom analyzer in my theme to support that kind of feature?

Thanks in advance,

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

**Comment by felipeelia** [comment]
Hi @OlgaLom, can you please share your ElasticPress Status Report and also the output of `wp elasticpress get-mapping --pretty`? Thanks!

**Comment by OlgaLom** [comment]
[elasticpress-report.txt](https://github.com/10up/ElasticPress/files/13364187/elasticpress-report.txt)
[post.txt](https://github.com/10up/ElasticPress/files/13364189/post.txt)
[term.txt](https://github.com/10up/ElasticPress/files/13364190/term.txt)

Hello @felipeelia and sorry for my late response, I didn't get any notification. 
the wp is not working for me, I used the curl command to get the results.

It is my first time working with Elastic so I'm a little bit lost with all the configuration

**Comment by felipeelia** [comment]
Hi @OlgaLom that was super useful, thanks. I gave it a try in my local environment, and it is working as expected (searching for "Darphin κρεμες" does return a post with "Darphin κρέμες". Do you happen to have any custom code related to ElasticPress added to your codebase?

**Comment by OlgaLom** [comment]
Hi, @felipeelia, yes I have, a lot. 

I have code related to the bellow topics

- Add custom taxonomies to be mapped
- Add custom fields to be mapped
- Add fields to the weight section
-  Add a new sorting option for search and pages (3 sorting fields)
- Add greeklish analyzer - Comment out 
- Change query fields - Comment out 

You can find all my code in the file.
[elastic_php.txt](https://github.com/10up/ElasticPress/files/13424738/elastic_php.txt)

I tried to disable all my code, reindex, and search "Darphin κρεμες" and "Darphin κρέμες" -- no results
With the code that I have sent you, I have these results
→ "Darphin κρεμες" -- no results
→ "Darphin κρέμες" -- some results. 

NOTE: excuse my code it is in a testing environment and it is not optimized. Some if statements and logic are terrible.

**Comment by felipeelia** [comment]
Hi @OlgaLom, 2 things:
1. Did you run a full sync (deleting all content) after commenting the mapping-related code? Just commenting it out won't take effect until you do that.
2. If you want your AJAX queries to use your weighting config, you'll need to add the ep_enable_do_weighting filter there. More [here](https://elasticpress.zendesk.com/hc/en-us/articles/22319421136141).

**Comment by OlgaLom** [comment]
Hello @felipeelia, we used the instant results popup feature and skipped the search page. The results seem fine for now, so we will see what the feature holds. :)

_Did you run a full sync (deleting all content) after commenting the mapping-related code? Just commenting it out won't take effect until you do that._
Yes, I did.

_If you want your AJAX queries to use your weighting config, you'll need to add the ep_enable_do_weighting filter there. More hehe_
I know about the weight configuration, but I don't think it is relevant to the query with no results.

Anyway, thanks for the info and your time, if I come across some problem relevant to the language I will open a new thread. 
Thanks again.
